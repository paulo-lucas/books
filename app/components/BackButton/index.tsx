import React from 'react';
import { StyleSheet } from 'react-native';
import { Icon, useTheme } from '@rneui/themed';
import { useNavigation } from '@react-navigation/native';

export const BackButton = () => {
  const { theme } = useTheme();
  const navigation = useNavigation();

  return (
    <Icon
      name="arrowleft"
      type="antdesign"
      color={theme.colors.primary}
      onPress={navigation.goBack}
      size={24}
      style={styles.arrow}
    />
  );
};

const styles = StyleSheet.create({
  arrow: {
    marginRight: 20,
    padding: 10,
  },
});
