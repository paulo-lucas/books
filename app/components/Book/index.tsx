import React from 'react';
import { View, Text, TouchableOpacity, StyleSheet } from 'react-native';
import { Image, useTheme, Skeleton } from '@rneui/themed';
import FavoriteButton from './FavoriteButton';
import BookSkeleton from './BookSkeleton';
import { useNavigation, NavigationProp } from '@react-navigation/native';
import type { RootStackParamList } from '@app/router/rootStackParams';

interface BookProps {
  refreshing: boolean;
  title?: string;
  thumbnail?: string;
  authors?: Array<string>;
  publisher?: string;
  identifier?: string;
}

export const Book: React.FC<BookProps> = React.memo(
  ({ refreshing, title, thumbnail, authors, publisher, identifier }) => {
    const navigation = useNavigation<NavigationProp<RootStackParamList>>();
    const { theme } = useTheme();
    const styles = createStyles(theme);

    if (refreshing) {
      return <BookSkeleton />;
    }

    const author = authors?.length ? authors[0] : publisher;
    const image = thumbnail?.replace('http://', 'https://');
    const id = identifier as string;

    const goToDetails = () => navigation.navigate('Details', { id });

    return (
      <TouchableOpacity style={styles.touchable} onPress={goToDetails}>
        <FavoriteButton style={styles.favorite} identifier={id} />
        <View style={styles.container}>
          <View style={styles.shadow}>
            <Image
              source={{
                uri: image ?? 'https://www.xy.com/images/placeholder.jpg',
              }}
              containerStyle={styles.image}
              PlaceholderContent={<Skeleton width={300} height={500} />}
            />
          </View>
          <Text numberOfLines={2} style={styles.title}>
            {title}
          </Text>
          <Text numberOfLines={2} style={styles.author}>
            {author}
          </Text>
        </View>
      </TouchableOpacity>
    );
  },
);

const createStyles = ({ colors }: any) =>
  StyleSheet.create({
    touchable: {
      width: '50%',
      position: 'relative',
    },
    favorite: {
      position: 'absolute',
      right: 30,
      zIndex: 1,
    },
    container: {
      width: '100%',
      paddingHorizontal: 10,
      paddingBottom: 20,
      paddingTop: 5,
      alignItems: 'center',
      justifyContent: 'center',
    },
    image: {
      width: '100%',
      height: undefined,
      aspectRatio: 76 / 130,
      borderRadius: 5,
      elevation: 20,
    },
    shadow: {
      shadowOffset: { width: -2, height: 4 },
      shadowOpacity: 0.5,
      shadowRadius: 3,
    },
    title: {
      fontSize: 15,
      fontWeight: '500',
      paddingLeft: 5,
      textAlign: 'center',
      marginTop: 10,
      color: colors.black,
    },
    author: {
      fontSize: 13,
      paddingLeft: 5,
      textAlign: 'center',
      color: colors.black,
    },
  });
