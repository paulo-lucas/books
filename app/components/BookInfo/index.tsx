import React from 'react';
import { Text, View, ScrollView, StyleSheet } from 'react-native';
import { Image, Skeleton } from '@rneui/themed';
import { useBookDetails } from '@app/hooks';
import { getBestFromImageLinks } from '@app/utils/getBestImageFromImagelinks';
import BookInfoSkeleton from './BookInfoSkeleton';

export const BookInfo = () => {
  const { book, isFavorite, loading, toggleFavorite } = useBookDetails();

  if (loading) {
    return <BookInfoSkeleton />;
  }

  const image = getBestFromImageLinks(book?.volumeInfo.imageLinks);
  console.log(image);

  return (
    <ScrollView style={styles.container}>
      <View style={styles.imageWrapper}>
        <Image
          source={{ uri: image }}
          style={styles.image}
          placeholderStyle={styles.placeholder}
          PlaceholderContent={<Skeleton style={styles.placeholder} />}
        />
      </View>

      <Text style={styles.title}>{book?.volumeInfo.title}</Text>
      <Text style={styles.author}>{book?.volumeInfo.authors.join(', ')}</Text>
      <Text style={styles.description}>{book?.volumeInfo.description}</Text>
    </ScrollView>
  );
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
    paddingHorizontal: 40,
    paddingVertical: 20,
  },
  imageWrapper: {
    width: '100%',
    alignItems: 'center',
  },
  image: {
    width: '80%',
    height: undefined,
    aspectRatio: 76 / 130,
  },
  placeholder: {
    width: '100%',
    height: '100%',
  },
  title: {
    marginTop: 20,
    fontSize: 24,
    width: '85%',
  },
  author: {
    marginTop: 10,
    marginBottom: 20,
    fontSize: 18,
    width: '60%',
  },
  description: {
    marginTop: 5,
    fontSize: 20,
    width: '100%',
  },
});
