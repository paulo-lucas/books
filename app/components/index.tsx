export * from './ToggleTheme';
export * from './Bookshelf';
export * from './SearchBar';
export * from './Filters';
