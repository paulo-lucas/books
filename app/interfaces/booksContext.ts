import type { Volume } from './volume';

export interface BooksState {
  total: number;
  itemsPerPage: number;
  page: number;
  data: Array<Volume>;
}

export interface BooksContextData {
  searchText: string;
  onChangeSearch(text: string): void;
  books: BooksState;
  refreshing: boolean;
  toggleOrderBy(): void;
  toggleFavoriteFilter(): void;
  toggleFavorite(isbn: string): void;
  isFavorite(favorite: string): boolean;
  onClearSearch(): void;
  refreshBooks(page?: number): void;
}

export interface UpdatePayload {
  type: 'update';
  page?: number;
  total: number;
  data: Array<Volume>;
}

export interface ClearPayload {
  type: 'clear';
}
