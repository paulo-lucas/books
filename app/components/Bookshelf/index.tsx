import React from 'react';
import { FlatList } from 'react-native';
import { Volume } from '@app/interfaces/volume';
import { Book } from '@app/components';
import Empty from './Empty';
import { useBooks } from '@app/hooks';

interface BookshelfProps {
  volumes?: Array<Volume>;
}

interface RenderItemArgs {
  item: Volume;
}

export const Bookshelf: React.FC<BookshelfProps> = () => {
  const { books, fetchBooks, refreshing } = useBooks();

  const renderItem = ({ item }: RenderItemArgs) => {
    return (
      <Book
        refreshing={refreshing}
        title={item.volumeInfo?.title}
        publisher={item.volumeInfo?.publisher}
        thumbnail={item.volumeInfo?.imageLinks.thumbnail}
        authors={item.volumeInfo?.authors}
        identifier={item.id}
      />
    );
  };

  return (
    <FlatList
      refreshing={refreshing}
      data={
        refreshing ? ([{}, {}, {}, {}, {}, {}] as Array<Volume>) : books.data
      }
      renderItem={renderItem}
      keyExtractor={(_, id) => String(id)}
      ListEmptyComponent={Empty}
      numColumns={2}
      onRefresh={fetchBooks}
    />
  );
};
