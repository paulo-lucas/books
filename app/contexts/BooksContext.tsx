import React, {
  createContext,
  useState,
  useReducer,
  PropsWithChildren,
  useEffect,
} from 'react';
import { BooksContextData } from '@app/interfaces/booksContext';
import { booksReducer, booksInitialState } from '@app/states/BooksReducer';
import { useDebounce } from '@app/hooks';
import { searchVolumes, storeFavorites, getFavorites } from '@app/services';
import { Alert } from 'react-native';
import { useCallback } from 'react';

export const BooksContext = createContext({} as BooksContextData);

export const BooksProvider: React.FC<PropsWithChildren> = ({ children }) => {
  const [books, dispatch] = useReducer(booksReducer, booksInitialState);
  const [searchText, setSearchText] = useState<string>('');
  const [refreshing, setRefreshing] = useState<boolean>(false);
  const [orderBy, setOrderBy] = useState<'relevance' | 'newest'>('relevance');
  const [favorites, setFavorites] = useState<Array<string>>([]);
  const [filterByFavorites, setFilterByFavorites] = useState<boolean>(false);

  const fetchFavorites = useCallback(async () => {
    const storageFavorites = await getFavorites();
    setFavorites(storageFavorites);
  }, []);

  const onClearSearch = () => dispatch({ type: 'clear' });

  const toggleFavoriteFilter = () => setFilterByFavorites(!filterByFavorites);

  const toggleOrderBy = () =>
    setOrderBy(orderBy === 'newest' ? 'relevance' : 'newest');

  const isFavorite = (isbn: string) => favorites.includes(isbn);

  const toggleFavorite = (isbn: string) => {
    let updatedFavorites;

    if (favorites.includes(isbn)) {
      updatedFavorites = favorites.filter(fav => fav !== isbn);
    } else {
      updatedFavorites = [...favorites, isbn];
    }

    setFavorites(updatedFavorites);
    storeFavorites(updatedFavorites);
  };

  const fetchBooks = useCallback(
    async (page: number, searchFromDebounce?: string) => {
      if (!searchText && !searchFromDebounce) {
        return onClearSearch();
      }

      const search = searchFromDebounce ?? searchText;
      const isbn = filterByFavorites ? favorites : [];
      const q = { search, isbn };
      const { itemsPerPage: maxResults } = books;
      const startIndex = page * maxResults;

      try {
        setRefreshing(true);
        const { data } = await searchVolumes({
          q,
          startIndex,
          maxResults,
          orderBy,
        });

        dispatch({
          type: 'update',
          total: data.totalItems,
          data: data.items ?? [],
        });

        setRefreshing(false);
      } catch (err) {
        Alert.alert('Could not retrieve books.');
      }
    },
    [books, favorites, filterByFavorites, orderBy, searchText],
  );

  const debounceSearch = useDebounce(fetchBooks, 1000);

  const onChangeSearch = (text: string) => {
    setSearchText(text);
    debounceSearch(0, text);
  };

  const refreshBooks = (page: number = 0) => {
    fetchBooks(page);
  };

  useEffect(() => {
    fetchFavorites();
  }, [fetchFavorites]);

  const contextData: BooksContextData = {
    books,
    refreshBooks,
    refreshing,
    searchText,
    onChangeSearch,
    onClearSearch,
    toggleOrderBy,
    toggleFavoriteFilter,
    isFavorite,
    toggleFavorite,
  };

  return (
    <BooksContext.Provider value={contextData}>
      {children}
    </BooksContext.Provider>
  );
};
