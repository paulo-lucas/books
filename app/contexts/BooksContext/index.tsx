import React, {
  createContext,
  useState,
  useReducer,
  PropsWithChildren,
  useEffect,
} from 'react';
import { BooksContextData } from '@app/interfaces/booksContext';
import { booksReducer, booksInitialState } from '@app/states/BooksReducer';
import { useDebounce } from '@app/hooks';
import { searchVolumes, storeFavorites, getFavorites } from '@app/services';
import { Alert } from 'react-native';
import { useCallback } from 'react';
import { VolumeSearchRequest } from '@app/interfaces/volume';

export const BooksContext = createContext({} as BooksContextData);

export const BooksProvider: React.FC<PropsWithChildren> = ({ children }) => {
  const [books, dispatch] = useReducer(booksReducer, booksInitialState);
  const [searchText, setSearchText] = useState<string>('');
  const [searchTextDebounce, setSearchTextDebounce] = useState<string>('');
  const [refreshing, setRefreshing] = useState<boolean>(false);
  const [orderBy, setOrderBy] = useState<'relevance' | 'newest'>('relevance');
  const [favorites, setFavorites] = useState<Array<string>>([]);
  const [filterByFavorites, setFilterByFavorites] = useState<boolean>(false);

  const fetchFavorites = useCallback(async () => {
    const favoritesFromStorage = await getFavorites();
    setFavorites(favoritesFromStorage);
  }, []);

  // Handle every request to list books, based on states
  const fetchBooks = useCallback(async () => {
    if (!searchTextDebounce && (!filterByFavorites || !favorites.length)) {
      return onClearSearch();
    }

    const identifier = filterByFavorites ? favorites : [];
    const q = { search: searchTextDebounce, identifier };
    const { itemsPerPage: maxResults, page } = books;
    const startIndex = page * maxResults;

    try {
      const request: VolumeSearchRequest = {
        q,
        startIndex,
        maxResults,
        orderBy,
        filter: 'partial',
        printType: 'books',
      };

      if (JSON.stringify(request) === books.request) {
        return;
      }

      setRefreshing(true);
      const { data } = await searchVolumes(request);

      const items = filterByFavorites
        ? data.items?.filter(item => favorites.includes(item.id))
        : data.items;

      dispatch({
        type: 'update',
        total: data.items?.length ?? 0,
        distinctTotal: items.length,
        data: items ?? [],
        page: page,
        request: JSON.stringify(request),
      });

      setRefreshing(false);
    } catch (err) {
      onClearSearch();
      setFilterByFavorites(false);
      Alert.alert('Could not retrieve books.');
    }
  }, [books, favorites, filterByFavorites, orderBy, searchTextDebounce]);

  const onClearSearch = () => dispatch({ type: 'clear' });

  const toggleOrderBy = () =>
    setOrderBy(orderBy === 'newest' ? 'relevance' : 'newest');

  // Toggle the filter, will make table show only favorited books
  const toggleFavoriteFilter = () => setFilterByFavorites(!filterByFavorites);

  // Toggle single book favorited / not favorited
  const toggleFavorite = (identifier: string) => {
    let updatedFavorites;

    if (favorites.includes(identifier)) {
      updatedFavorites = favorites.filter(fav => fav !== identifier);
    } else {
      updatedFavorites = [...favorites, identifier];
    }

    updatedFavorites = updatedFavorites.filter(fav => !!fav);

    setFavorites(updatedFavorites);
    storeFavorites(updatedFavorites);
  };

  const isFavorite = (identifier: string) => favorites.includes(identifier);

  const debounceSearch = useDebounce(setSearchTextDebounce, 500);

  const onChangeSearch = (text: string) => {
    setSearchText(text);
    debounceSearch(text);
  };

  const onChangePage = (page: number) => dispatch({ type: 'moveToPage', page });

  useEffect(() => {
    fetchBooks();
  }, [fetchBooks]);

  useEffect(() => {
    fetchFavorites();
  }, [fetchFavorites]);

  const contextData: BooksContextData = {
    books,
    refreshing,
    searchText,
    onChangeSearch,
    onClearSearch,
    fetchBooks,
    toggleOrderBy,
    toggleFavoriteFilter,
    orderBy,
    filterByFavorites,
    isFavorite,
    toggleFavorite,
    onChangePage,
  };

  return (
    <BooksContext.Provider value={contextData}>
      {children}
    </BooksContext.Provider>
  );
};
