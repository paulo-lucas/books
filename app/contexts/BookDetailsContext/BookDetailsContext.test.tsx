import 'react-native';
import React, { ReactElement } from 'react';
import { render, RenderOptions, screen } from '../../../tests/test-utils';
import { BookDetailsProvider } from './';
import { BooksContext } from '../BooksContext';
import { BooksContextData } from '@app/interfaces/booksContext';
import { BookInfo } from '@app/components';
jest.useFakeTimers();

const mockGoBack = jest.fn();

const isFavorite = jest.fn();
const toggleFavorite = jest.fn();

jest.mock('@react-navigation/native', () => {
  const actualRNavigation = jest.requireActual('@react-navigation/native');
  return {
    ...actualRNavigation,
    useRoute: () => ({
      params: {},
    }),
    useNavigation: () => ({
      navigation: { goBack: mockGoBack },
    }),
  };
});

const customRender = (ui: ReactElement, options: RenderOptions = {}) => {
  return render(
    <BooksContext.Provider
      value={{ isFavorite, toggleFavorite } as unknown as BooksContextData}>
      <BookDetailsProvider>{ui}</BookDetailsProvider>
    </BooksContext.Provider>,
    options,
  );
};

describe('details context', () => {
  it('should render correctly with consumer', async () => {
    customRender(<BookInfo />);

    expect(screen.toJSON()).toMatchSnapshot();
  });
});
