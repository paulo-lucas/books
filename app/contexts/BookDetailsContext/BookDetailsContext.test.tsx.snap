// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`details context should render correctly with consumer 1`] = `
<RNCSafeAreaView
  edges={
    Array [
      "bottom",
      "left",
      "right",
      "top",
    ]
  }
  style={
    Object {
      "backgroundColor": undefined,
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "flex": 1,
        "paddingHorizontal": 20,
        "paddingVertical": 20,
      }
    }
  >
    <View
      accessibilityLabel="loading..."
      accessibilityRole="none"
      accessible={false}
      onLayout={[Function]}
      replaceTheme={[Function]}
      style={
        Array [
          Object {
            "borderRadius": 2,
            "overflow": "hidden",
          },
          Object {
            "backgroundColor": "#bdc6cf",
            "height": 12,
            "width": "100%",
          },
          undefined,
          Object {
            "alignSelf": "center",
            "aspectRatio": 0.5846153846153846,
            "height": undefined,
            "width": "80%",
          },
        ]
      }
      testID="RNE__Skeleton"
      updateTheme={[Function]}
    >
      <View
        collapsable={false}
        style={
          Object {
            "backgroundColor": "rgb(166, 178, 190)",
            "height": "100%",
            "opacity": 1,
            "width": "100%",
          }
        }
      />
    </View>
    <View
      accessibilityLabel="loading..."
      accessibilityRole="none"
      accessible={false}
      onLayout={[Function]}
      replaceTheme={[Function]}
      style={
        Array [
          Object {
            "borderRadius": 2,
            "overflow": "hidden",
          },
          Object {
            "backgroundColor": "#bdc6cf",
            "height": 12,
            "width": "100%",
          },
          undefined,
          Object {
            "height": 24,
            "marginTop": 20,
            "width": "85%",
          },
        ]
      }
      testID="RNE__Skeleton"
      updateTheme={[Function]}
    >
      <View
        collapsable={false}
        style={
          Object {
            "backgroundColor": "rgb(166, 178, 190)",
            "height": "100%",
            "opacity": 1,
            "width": "100%",
          }
        }
      />
    </View>
    <View
      accessibilityLabel="loading..."
      accessibilityRole="none"
      accessible={false}
      onLayout={[Function]}
      replaceTheme={[Function]}
      style={
        Array [
          Object {
            "borderRadius": 2,
            "overflow": "hidden",
          },
          Object {
            "backgroundColor": "#bdc6cf",
            "height": 12,
            "width": "100%",
          },
          undefined,
          Object {
            "height": 18,
            "marginBottom": 20,
            "marginTop": 10,
            "width": "60%",
          },
        ]
      }
      testID="RNE__Skeleton"
      updateTheme={[Function]}
    >
      <View
        collapsable={false}
        style={
          Object {
            "backgroundColor": "rgb(166, 178, 190)",
            "height": "100%",
            "opacity": 1,
            "width": "100%",
          }
        }
      />
    </View>
    <View
      accessibilityLabel="loading..."
      accessibilityRole="none"
      accessible={false}
      onLayout={[Function]}
      replaceTheme={[Function]}
      style={
        Array [
          Object {
            "borderRadius": 2,
            "overflow": "hidden",
          },
          Object {
            "backgroundColor": "#bdc6cf",
            "height": 12,
            "width": "100%",
          },
          undefined,
          Object {
            "height": 20,
            "marginTop": 5,
            "width": "100%",
          },
        ]
      }
      testID="RNE__Skeleton"
      updateTheme={[Function]}
    >
      <View
        collapsable={false}
        style={
          Object {
            "backgroundColor": "rgb(166, 178, 190)",
            "height": "100%",
            "opacity": 1,
            "width": "100%",
          }
        }
      />
    </View>
    <View
      accessibilityLabel="loading..."
      accessibilityRole="none"
      accessible={false}
      onLayout={[Function]}
      replaceTheme={[Function]}
      style={
        Array [
          Object {
            "borderRadius": 2,
            "overflow": "hidden",
          },
          Object {
            "backgroundColor": "#bdc6cf",
            "height": 12,
            "width": "100%",
          },
          undefined,
          Object {
            "height": 20,
            "marginTop": 5,
            "width": "100%",
          },
        ]
      }
      testID="RNE__Skeleton"
      updateTheme={[Function]}
    >
      <View
        collapsable={false}
        style={
          Object {
            "backgroundColor": "rgb(166, 178, 190)",
            "height": "100%",
            "opacity": 1,
            "width": "100%",
          }
        }
      />
    </View>
    <View
      accessibilityLabel="loading..."
      accessibilityRole="none"
      accessible={false}
      onLayout={[Function]}
      replaceTheme={[Function]}
      style={
        Array [
          Object {
            "borderRadius": 2,
            "overflow": "hidden",
          },
          Object {
            "backgroundColor": "#bdc6cf",
            "height": 12,
            "width": "100%",
          },
          undefined,
          Object {
            "height": 20,
            "marginTop": 5,
            "width": "100%",
          },
        ]
      }
      testID="RNE__Skeleton"
      updateTheme={[Function]}
    >
      <View
        collapsable={false}
        style={
          Object {
            "backgroundColor": "rgb(166, 178, 190)",
            "height": "100%",
            "opacity": 1,
            "width": "100%",
          }
        }
      />
    </View>
    <View
      accessibilityLabel="loading..."
      accessibilityRole="none"
      accessible={false}
      onLayout={[Function]}
      replaceTheme={[Function]}
      style={
        Array [
          Object {
            "borderRadius": 2,
            "overflow": "hidden",
          },
          Object {
            "backgroundColor": "#bdc6cf",
            "height": 12,
            "width": "100%",
          },
          undefined,
          Object {
            "height": 20,
            "marginTop": 5,
            "width": "100%",
          },
        ]
      }
      testID="RNE__Skeleton"
      updateTheme={[Function]}
    >
      <View
        collapsable={false}
        style={
          Object {
            "backgroundColor": "rgb(166, 178, 190)",
            "height": "100%",
            "opacity": 1,
            "width": "100%",
          }
        }
      />
    </View>
    <View
      accessibilityLabel="loading..."
      accessibilityRole="none"
      accessible={false}
      onLayout={[Function]}
      replaceTheme={[Function]}
      style={
        Array [
          Object {
            "borderRadius": 2,
            "overflow": "hidden",
          },
          Object {
            "backgroundColor": "#bdc6cf",
            "height": 12,
            "width": "100%",
          },
          undefined,
          Object {
            "height": 20,
            "marginTop": 5,
            "width": "100%",
          },
        ]
      }
      testID="RNE__Skeleton"
      updateTheme={[Function]}
    >
      <View
        collapsable={false}
        style={
          Object {
            "backgroundColor": "rgb(166, 178, 190)",
            "height": "100%",
            "opacity": 1,
            "width": "100%",
          }
        }
      />
    </View>
    <View
      accessibilityLabel="loading..."
      accessibilityRole="none"
      accessible={false}
      onLayout={[Function]}
      replaceTheme={[Function]}
      style={
        Array [
          Object {
            "borderRadius": 2,
            "overflow": "hidden",
          },
          Object {
            "backgroundColor": "#bdc6cf",
            "height": 12,
            "width": "100%",
          },
          undefined,
          Object {
            "height": 20,
            "marginTop": 5,
            "width": "70%",
          },
        ]
      }
      testID="RNE__Skeleton"
      updateTheme={[Function]}
    >
      <View
        collapsable={false}
        style={
          Object {
            "backgroundColor": "rgb(166, 178, 190)",
            "height": "100%",
            "opacity": 1,
            "width": "100%",
          }
        }
      />
    </View>
  </View>
</RNCSafeAreaView>
`;
